### PR TITLE
fixed bug in tensorflow dot when multiple dims are dynamic

### PR DESCRIPTION
**Pull request by linxihui**
fixed failure at `K.dot` when `shape(x) = (None, None, 9)` (could appear in `recurrent`).

### CODE REVIEW

**Comment by fchollet** [comment]
Please add a unit test to check for this case (in tests/keras/backend)

File: tests/keras/backend/test_backends.py [hunk 1]
```diff
@@ -9,32 +9,60 @@
 from keras.utils.np_utils import convert_kernel
 
 
-def check_single_tensor_operation(function_name, input_shape, **kwargs):
-    val = np.random.random(input_shape) - 0.5
-    xth = KTH.variable(val)
-    xtf = KTF.variable(val)
+def check_single_tensor_operation(function_name, input_shape, test_input_shape=None, **kwargs):
```
Comment (fchollet) [inline tests/keras/backend/test_backends.py]: ?

Comment (linxihui) [inline tests/keras/backend/test_backends.py]: The interface to allow dynamic length in shape like `input_shape=(None, 3)` (to construct the tensor). The user can specify the actual `test_input_shape` in test time when `input_shape` has dynamic length in one dimension. see example [here](https://github.com/linxihui/keras/blob/7965fe76e65378e71b391732026b0073034ac45b/tests/keras/backend/test_backends.py#L99). The existent test function does not accept `input_shape=(None, 3)`.

**Comment by fchollet** [comment]
I wasn't happy with your API changes in the test utils, so I didn't merge this PR. I did however recently commit your fix, with your email as author.

**Pull request closed without merging**